MapboxCoreNavigation/FreeDriveLocationManager.swift

```diff
@@ -0,0 +1,196 @@
+    }
+
+    func setCustomLocation(_ location: CLLocation?) {
+        guard let location = location else { return }
+        stopUpdatingLocation()
+        stopUpdatingHeading()
+        proxyDelegate.locationManager(self, didUpdateLocations: [location])
+    }
+
+    // MARK: MGLLocationManager
+
+    override public var delegate: CLLocationManagerDelegate? {
+        get {
+            proxyDelegate.delegate
+        }
+        set {
+            proxyDelegate.delegate = newValue
+        }
+    }
+
+    public func debugView(onUpdated: ((CLLocationCoordinate2D, CLLocationCoordinate2D)->Void)?) -> UIView {
+        let debugView = FreeDriveDebugInfoView()
+        proxyDelegate.debugDelegate = debugView
+        debugView.onUpdated = onUpdated
+        return debugView
+    }
+
+    private class ProxyDelegate: NSObject, CLLocationManagerDelegate {
+        private var navNative: Navigator!
+        var delegate: CLLocationManagerDelegate?
+        weak var debugDelegate: FreeDriveDebugInfoListener?
+
+        required init(navigator: Navigator? = nil) {
+            self.navNative = navigator ?? Navigator()
+            super.init()
+        }
+
+        deinit {
+            self.navNative = nil
+        }
+
+        // MARK: CLLocationManagerDelegate
+
+        public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+            if let loc = locations.first {
+                navNative?.updateLocation(for: FixLocation(loc))
+                let projectedDate = Date()
+                let status = navNative.getStatusForTimestamp(projectedDate)
+                delegate?.locationManager?(manager, didUpdateLocations: [CLLocation(status.location)])
+
+                debugDelegate?.didGet(location: CLLocation(status.location), with: status.map_matcher_output.matches, for: loc)
+            }
+        }
+
+        public func locationManager(_ manager: CLLocationManager, didUpdateHeading newHeading: CLHeading) {
+            delegate?.locationManager?(manager, didUpdateHeading: newHeading)
+        }
+
+        public func locationManagerShouldDisplayHeadingCalibration(_ manager: CLLocationManager) -> Bool {
+            return delegate?.locationManagerShouldDisplayHeadingCalibration?(manager) ?? false
+        }
+
+        public func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
+            delegate?.locationManager?(manager, didFailWithError: error)
+        }
+    }
+}
+
+private class Accounts: SkuTokenSource {
+    init() {
+        MBXAccounts.activateSKUID(.navigationUser)
+    }
+    func getToken() -> String {
+        MBXAccounts.skuToken
+    }
+
+    var peer: MBXPeerWrapper?
+}
+
+protocol FreeDriveDebugInfoListener: AnyObject {
+    func didGet(location: CLLocation, with matches: [MapMatch], for rawLocation: CLLocation)
+}
+
+class FreeDriveDebugInfoView: UIView, FreeDriveDebugInfoListener, UITableViewDataSource, UITableViewDelegate {
```

Comment:
If the intention is to ship this debugging view with the SDK, it needs to be moved into MapboxNavigation. MapboxCoreNavigation is for non-UI components only. There should be no dependency on UIKit.

Reply:
moved to `MapboxNavigation`